### PR TITLE
Optimize/output/v2

### PR DESCRIPTION
**Pull request by victorjulien**
Describe changes:
- Various optimizations to output framework.
- @jasonish can you check they changes to the 'root loggers'?
- fixes #4374. Unix socket was broken.

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):
- PR victorjulien-pcap: https://buildbot.openinfosecfoundation.org/builders/victorjulien-pcap/builds/340
- PR victorjulien: https://buildbot.openinfosecfoundation.org/builders/victorjulien/builds/341

### CODE REVIEW

File: src/output.c [hunk 1]
```diff
@@ -1015,9 +1023,46 @@ void OutputRegisterRootLogger(ThreadInitFunc ThreadInit,
...
     logger->ThreadExitPrintStats = ThreadExitPrintStats;
     logger->LogFunc = LogFunc;
+    logger->ActiveCntFunc = ActiveCntFunc;
     TAILQ_INSERT_TAIL(&registered_loggers, logger, entries);
 }
 
+static void OutputRegisterActiveLogger(RootLogger *reg)
+{
+    RootLogger *logger = SCCalloc(1, sizeof(*logger));
+    if (logger == NULL) {
```
Comment (jlucovsky) [inline src/output.c]: Suggest logging an error message and terminating. This shouldn't happen, especially early in the execution lifetime of suri.

Comment (victorjulien) [reply src/output.c]: Good point. Making this a fatal error.

**Comment by jasonish** [comment]
Makes sense. Looks good.

**Comment by victorjulien** [comment]
Will repost this when 6.0dev opens.

**Pull request closed without merging**